<a href="https://colab.research.google.com/github/vishalkarur08/Hypo_testing-/blob/main/Assignment_04_HYPO_VK.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Task
Perform a hypothesis test to investigate the restaurant owners' claim that the weekly operating costs are higher than the theoretical model suggests. Follow the provided steps: state hypotheses, calculate the test statistic, determine the critical value, make a decision, and state a conclusion. Use the given data: theoretical cost model W = $1,000 + $5X, sample mean weekly cost of Rs. 3,050 from 25 restaurants, and X follows a normal distribution with mean 600 and standard deviation 25. Use an alpha level of 5%.

## State the hypotheses

### Subtask:
Define the null and alternative hypotheses for the test.


**Reasoning**:
Define the null and alternative hypotheses based on the problem description.



In [ ]:
# The theoretical cost model is W = $1,000 + $5X
# X follows a normal distribution with mean 600 and standard deviation 25
# The theoretical mean weekly operating cost is E[W] = E[1000 + 5X] = 1000 + 5 * E[X]
theoretical_mean_cost = 1000 + 5 * 600

# Null Hypothesis (H0): The mean weekly operating cost is equal to the theoretical mean cost.
# H0: mu = theoretical_mean_cost

# Alternative Hypothesis (H1): The mean weekly operating cost is higher than the theoretical mean cost.
# H1: mu > theoretical_mean_cost

print(f"Null Hypothesis (H0): The mean weekly operating cost is equal to ${theoretical_mean_cost}")
print(f"Alternative Hypothesis (H1): The mean weekly operating cost is higher than ${theoretical_mean_cost}")

Null Hypothesis (H0): The mean weekly operating cost is equal to $4000
Alternative Hypothesis (H1): The mean weekly operating cost is higher than $4000


## Calculate the test statistic

### Subtask:
Calculate the test statistic using the provided formula and data.


**Reasoning**:
Calculate the test statistic based on the provided formula and data.



In [ ]:
import numpy as np

# Sample mean weekly cost
sample_mean_cost = 3050

# Sample size
n = 25

# Standard deviation of X
std_dev_X = 25

# Calculate the theoretical standard deviation of the weekly operating cost (W)
# Since W = 1000 + 5X, the standard deviation of W is 5 * std_dev_X
theoretical_std_dev_W = 5 * std_dev_X

# Calculate the standard error of the sample mean weekly cost
# Standard error = theoretical_std_dev_W / sqrt(n)
standard_error = theoretical_std_dev_W / np.sqrt(n)

# Calculate the test statistic (Z-score)
# Z = (sample_mean - theoretical_mean) / standard_error
test_statistic = (sample_mean_cost - theoretical_mean_cost) / standard_error

print(f"Theoretical standard deviation of W: {theoretical_std_dev_W}")
print(f"Standard error of the sample mean: {standard_error}")
print(f"Test statistic (Z-score): {test_statistic}")

Theoretical standard deviation of W: 125
Standard error of the sample mean: 25.0
Test statistic (Z-score): -38.0


## Determine the critical value

### Subtask:
Find the critical value from the standard normal distribution table for the given alpha level.


**Reasoning**:
Find the critical value for a one-tailed test with alpha = 0.05.



In [ ]:
from scipy.stats import norm

# Since this is a one-tailed test (H1: mu > theoretical_mean_cost) and the alpha level is 5% (0.05),
# we find the Z-score corresponding to a cumulative probability of 1 - 0.05 = 0.95 in the standard normal distribution table.
critical_value = norm.ppf(0.95)

print(f"Critical value: {critical_value}")

Critical value: 1.6448536269514722


## Make a decision

### Subtask:
Compare the test statistic and critical value to decide whether to reject the null hypothesis.


**Reasoning**:
Compare the test statistic and critical value and determine whether to reject the null hypothesis.



In [ ]:
# Compare the test statistic with the critical value
# For a right-tailed test, we reject H0 if test_statistic > critical_value
if test_statistic > critical_value:
    decision = "Reject H0"
else:
    decision = "Fail to reject H0"

# Print the decision
print(f"Test Statistic: {test_statistic}")
print(f"Critical Value: {critical_value}")
print(f"Decision: {decision}")

Test Statistic: -38.0
Critical Value: 1.6448536269514722
Decision: Fail to reject H0


## State the conclusion

### Subtask:
Based on the decision, write a conclusion about the restaurant owners' claim.


**Reasoning**:
Write the conclusion based on the decision to fail to reject the null hypothesis, stating whether there is sufficient evidence to support the claim and mentioning the relationship between the test statistic and critical value.



In [ ]:
# Based on the decision to "Fail to reject H0", write the conclusion.

conclusion = f"""
Conclusion:

Based on a one-tailed hypothesis test at the 5% significance level, the test statistic (-38.0) is less than the critical value (1.645). Since the test statistic does not fall in the rejection region (test statistic > critical value), we fail to reject the null hypothesis.

There is not sufficient statistical evidence at the 5% significance level to support the restaurant owners' claim that the weekly operating costs are higher than the theoretical model suggests.
"""

print(conclusion)


Conclusion:

Based on a one-tailed hypothesis test at the 5% significance level, the test statistic (-38.0) is less than the critical value (1.645). Since the test statistic does not fall in the rejection region (test statistic > critical value), we fail to reject the null hypothesis.

There is not sufficient statistical evidence at the 5% significance level to support the restaurant owners' claim that the weekly operating costs are higher than the theoretical model suggests.



## Summary:

### Data Analysis Key Findings

*   The theoretical mean weekly operating cost based on the model \(W = \$1,000 + \$5X\) and \(E[X] = 600\) is calculated to be \$4,000.
*   The standard deviation of the weekly operating cost (W) is calculated as 5 times the standard deviation of X, which is \$125.
*   The standard error of the sample mean weekly cost for a sample of 25 restaurants is calculated as \$25.
*   The calculated test statistic (Z-score) is -38.0.
*   For a one-tailed test with an alpha level of 5%, the critical value from the standard normal distribution is approximately 1.645.
*   Since the test statistic (-38.0) is less than the critical value (1.645), the decision is to fail to reject the null hypothesis.

### Insights or Next Steps

*   The sample mean (\$3,050) is significantly lower than the theoretical mean (\$4,000), which is contrary to the restaurant owners' claim of higher costs.
*   Investigate potential reasons for the discrepancy between the sample mean and the theoretical mean, such as inaccuracies in the theoretical model or data collection issues.
